src/gt4py/next/iterator/transforms/trace_shifts.py

```diff
@@ -12,15 +12,19 @@
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+import types
 from collections.abc import Callable
 from dataclasses import dataclass
-from typing import Any, Final
+from typing import Any, Final, List, Union
 
 from gt4py.eve import NodeTranslator
 from gt4py.next.iterator import ir
 from gt4py.next.iterator.transforms.collect_shifts import ALL_NEIGHBORS
 
 
+VALUE_TOKEN = types.new_class("VALUE_TOKEN")
```

Comment:
This is just nitpicking but why is it created as a class using `types.new_class` instead of just declaring a new object? I mean, if this value is always only checked by identity as a sentinel, it is common to use an object:
```suggestion
VALUE_TOKEN = object()
```

Reply:
Occasionally something breaks and it is very useful to have a the information of what type of object, e.g. a `ValueToken`, directly visible in the debugger. Does that make sense?

Reply:
Ok, fine for me. 
BTW, in case you didn't know, there is a `NOTHING` sentinel class in `eve.type_definitions` with its own metaclass to evaluate as `False` bool.